server/src/providers/project/index.ts

```diff
@@ -40,34 +45,60 @@ export async function initialise() {
 }
 
 async function loadWorkspace() {
-	const uris = await getProjectFiles();
+	const workspaces = await connection.workspace.getWorkspaceFolders();
+
+	console.log(workspaces);
+
+	if (workspaces) {
+		let uris: string[] = [];
+
+		workspaces.forEach((workspaceUri => {
+			const folderPath = URI.parse(workspaceUri.uri).path;
+
+			console.log(`Starting search of: ${folderPath}`);
+			const files = glob.sync(projectFilesGlob, {
+				cwd: folderPath,
+				absolute: true,
+				nocase: true,
+			});
+
+			console.log(`Found RPGLE files: ${files.length}`);
 
-	if (uris) {
-		await Promise.allSettled(uris?.map(uri => loadFile(uri)));
+			uris.push(...files.map(file => URI.from({
+				scheme: `file`,
+				path: file
+			}).toString()))
+		}));
+
+		if (uris.length < 1000) {
+			for (const uri of uris) {
+				await loadLocalFile(uri);
+			}
+		} else {
+			console.log(`Disabling project mode for large project.`);
+			isEnabled = false;
+		}
 	}
 }
 
-async function loadFile(uri: string) {
-	const content = await getFileRequest(uri);
-
-	if (content) {
-		parser.getDocs(uri, content).then(cache => {
-			// Linter / reference collector only works on free-format.
-			if (cache) {
-				if (content.length >= 6 && content.substring(0, 6).toUpperCase() === `**FREE`) {
-					Linter.getErrors({
-						uri,
-						content,
-					}, {
-						CollectReferences: true
-					}, cache);
-				}
-			}
-		});
+async function loadLocalFile(uri: string) {
+	const validPath = URI.parse(uri).path;
+	const content = await fs.readFile(validPath, { encoding: `utf-8` });
```

Comment:
We stopped using getFileRequest here to remove a call back to the client extension.